internal/venice-client-common/src/main/java/com/linkedin/venice/schema/SchemaData.java

```diff
@@ -41,17 +39,17 @@ public final class SchemaData {
 
   public SchemaData(String storeName) {
     this.storeName = storeName;
-    valueSchemaMap = new TreeMap<>();
-    valueSchemaRMap = new VeniceConcurrentHashMap<>();
+    this.valueSchemaMap = new ConcurrentSkipListMap<>();
+    this.valueSchemaRMap = new VeniceConcurrentHashMap<>();
 
-    updateSchemaMap = new VeniceConcurrentHashMap<>();
-    updateSchemaRMap = new VeniceConcurrentHashMap<>();
+    this.updateSchemaMap = new VeniceConcurrentHashMap<>();
+    this.updateSchemaRMap = new VeniceConcurrentHashMap<>();
 
-    rmdSchemaMap = new VeniceConcurrentHashMap<>();
-    rmdSchemaRMap = new VeniceConcurrentHashMap<>();
+    this.rmdSchemaMap = new VeniceConcurrentHashMap<>();
+    this.rmdSchemaRMap = new VeniceConcurrentHashMap<>();
 
-    updateSchemaExistenceSet = new HashSet<>();
-    rmdSchemaExistenceSet = new HashSet<>();
+    this.updateSchemaExistenceSet = ConcurrentHashMap.newKeySet();
```

Comment:
Can we use `ConcurrentSkipListMap` here too till we stop supporting JDK8.

Since this PR already touches routers, and if you're feeling frisky, there is a function (`Utils#newConcurrentSet`) which also has the potential of running into the same JDK8 issue. This function only has one usage, and it is in routers. We could remove that usage too in this PR

Reply:
But sets don't have `computeIfAbsent` so I think the JDK8 issue is moot for them, right?

Reply:
I vaguely remember that the root cause of the race condition was affecting the sets too. I can check one more

Reply:
Hmm... but adding to a set "onlyIfAbsent" sounds a bit odd 😅 ...? The only situations I can think of where that would matter seem pretty fingy...

Reply:
Oh I see... add always calls putVal with this param set to true... that's weird. Oh well.

Reply:
But the thing is... `Utils::newConcurrentSet` returns `VeniceConcurrentHashMap::newKeySet` which itself is just `ConcurrentHashMap::newKeySet`...

Reply:
Yeah. I meant getting rid of `Utils::newConcurrentSet` too because it has the same issue and replacing it with `ConcurrentSkipListSet`

Reply:
Ok i see... sure, I can do that.